src/http.c

```diff
@@ -42,10 +45,69 @@ addReplyHTTPHeader (
   sdsfree(http_header);
 }
 
-void
-processHTTPRequest (
-  client               *c
-) {
+void readHTTPRequest (aeEventLoop *el, int fd, void *privdata, int mask) {
+  UNUSED(el);
+  UNUSED(mask);
+  struct HttpClient *pclient = (struct HttpClient*)privdata;
+  /* BUGBUG This is just a proof of concept for the reading code, I'm sure there's a better way */
+  if ((pclient->cbQueryAlloc - pclient->cbQuery) < HTTP_READ_CHUNK)
+  {
+    pclient->cbQueryAlloc += HTTP_READ_CHUNK; // REVIEW: Growth limits?
+    pclient->szQuery = zrealloc(pclient->szQuery, pclient->cbQueryAlloc, MALLOC_LOCAL); // REVIEW handle NULL
+  }
+  ssize_t cbRead = read(fd, pclient->szQuery + pclient->cbQuery, HTTP_READ_CHUNK-1);
+  if (cbRead < 0)
+  {
+    if (errno != EAGAIN)
+      perror("Failed to read from HTTP socket");
+    serverAssert(errno == EAGAIN);  // REVIEW: Handle other errors
+    return;
+  }
+  else if (cbRead > 0)
+  {
+    pclient->cbQuery += cbRead;
+    pclient->szQuery[pclient->cbQuery] = '\0';
+    processHTTPRequest(el, pclient);
+  }
+}
+
+#define MAX_ACCEPTS_PER_CALL 1000
+void acceptHttpTcpHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
+  int cport, cfd, max = MAX_ACCEPTS_PER_CALL;
+  char cip[NET_IP_STR_LEN];
+  UNUSED(mask);
+  UNUSED(privdata);
+
+  while(max--) 
+  {
+    cfd = anetTcpAccept(server.neterr, fd, cip, sizeof(cip), &cport);
+    if (cfd == ANET_ERR) {
+        if (errno != EWOULDBLOCK)
+            serverLog(LL_WARNING,
+                "Accepting HTTP connection: %s", server.neterr);
+        return;
+    }
+    serverLog(LL_VERBOSE,"HTTP Accepted %s:%d", cip, cport);
+
+    struct HttpClient *pclient = (struct HttpClient*)zmalloc(sizeof(struct HttpClient), MALLOC_LOCAL);
```

Comment:
FYI MALLOC_LOCAL means keep in RAM not FLASH (if the FLASH feature is used).  Probably could be a better name.